From b3cf2ee42f0799e190c875f3af2ce6e5a7e287ce Mon Sep 17 00:00:00 2001
From: Tim Parenti <tim@timtimeonline.com>
Date: Sat, 3 Oct 2020 17:58:38 -0400
Subject: [PATCH 40/47] Remove obsolete file systemv

* systemv: Remove obsolete file.
* zishrink.awk (BEGIN): Don't depend on the old file.
(process_input_line): Remove exception for the Rules the old file contains.
* Makefile (NDATA, VERSION_DEPS): Remove all uses.
* NEWS: Document this.
---
 Makefile     |  4 ++--
 NEWS         |  5 +++++
 systemv      | 39 ---------------------------------------
 zishrink.awk |  4 ----
 4 files changed, 7 insertions(+), 45 deletions(-)
 delete mode 100644 systemv

diff --git a/Makefile b/Makefile
index 8abd4d0..0a06521 100644
--- a/Makefile
+++ b/Makefile
@@ -533,7 +533,7 @@ DOCS=		$(MANS) date.1 $(MANTXTS) $(WEB_PAGES)
 PRIMARY_YDATA=	africa antarctica asia australasia \
 		europe northamerica southamerica
 YDATA=		$(PRIMARY_YDATA) etcetera
-NDATA=		systemv factory
+NDATA=		factory
 TDATA_TO_CHECK=	$(YDATA) $(NDATA) backward pacificnew
 TDATA=		$(YDATA) $(NDATA) $(BACKWARD)
 ZONETABLES=	zone1970.tab zone.tab
@@ -569,7 +569,7 @@ VERSION_DEPS= \
 		leap-seconds.list leapseconds.awk localtime.c \
 		newctime.3 newstrftime.3 newtzset.3 northamerica \
 		pacificnew private.h \
-		southamerica strftime.c systemv theory.html \
+		southamerica strftime.c theory.html \
 		time2posix.3 tz-art.html tz-how-to.html tz-link.html \
 		tzfile.5 tzfile.h tzselect.8 tzselect.ksh \
 		workman.sh yearistype.sh \
diff --git a/NEWS b/NEWS
index f0858f8..2cabf90 100644
--- a/NEWS
+++ b/NEWS
@@ -58,6 +58,11 @@ Unreleased, experimental changes
     The Makefile now defaults POSIXRULES to '-', so the posixrules
     feature (obsolete as of 2019b) is no longer installed by default.
 
+  Changes to documentation and commentary
+
+    The long-obsolete file systemv has been removed from the
+    distribution.  (Thanks to Tim Parenti.)
+
 
 Release 2020a - 2020-04-23 16:03:47 -0700
 
diff --git a/systemv b/systemv
deleted file mode 100644
index a8c037c..0000000
--- a/systemv
+++ /dev/null
@@ -1,39 +0,0 @@
-# tzdb data for System V rules (this file is obsolete)
-
-# This file is in the public domain, so clarified as of
-# 2009-05-17 by Arthur David Olson.
-
-# Old rules, should the need arise.
-# No attempt is made to handle Newfoundland, since it cannot be expressed
-# using the System V "TZ" scheme (half-hour offset), or anything outside
-# North America (no support for non-standard DST start/end dates), nor
-# the changes in the DST rules in the US after 1976 (which occurred after
-# the old rules were written).
-#
-# If you need the old rules, uncomment ## lines.
-# Compile this *without* leap second correction for true conformance.
-
-# Rule	NAME	FROM	TO	TYPE	IN	ON	AT	SAVE	LETTER/S
-Rule	SystemV	min	1973	-	Apr	lastSun	2:00	1:00	D
-Rule	SystemV	min	1973	-	Oct	lastSun	2:00	0	S
-Rule	SystemV	1974	only	-	Jan	6	2:00	1:00	D
-Rule	SystemV	1974	only	-	Nov	lastSun	2:00	0	S
-Rule	SystemV	1975	only	-	Feb	23	2:00	1:00	D
-Rule	SystemV	1975	only	-	Oct	lastSun	2:00	0	S
-Rule	SystemV	1976	max	-	Apr	lastSun	2:00	1:00	D
-Rule	SystemV	1976	max	-	Oct	lastSun	2:00	0	S
-
-# Zone	NAME		STDOFF	RULES/SAVE	FORMAT	[UNTIL]
-## Zone	SystemV/AST4ADT	-4:00	SystemV		A%sT
-## Zone	SystemV/EST5EDT	-5:00	SystemV		E%sT
-## Zone	SystemV/CST6CDT	-6:00	SystemV		C%sT
-## Zone	SystemV/MST7MDT	-7:00	SystemV		M%sT
-## Zone	SystemV/PST8PDT	-8:00	SystemV		P%sT
-## Zone	SystemV/YST9YDT	-9:00	SystemV		Y%sT
-## Zone	SystemV/AST4	-4:00	-		AST
-## Zone	SystemV/EST5	-5:00	-		EST
-## Zone	SystemV/CST6	-6:00	-		CST
-## Zone	SystemV/MST7	-7:00	-		MST
-## Zone	SystemV/PST8	-8:00	-		PST
-## Zone	SystemV/YST9	-9:00	-		YST
-## Zone	SystemV/HST10	-10:00	-		HST
diff --git a/zishrink.awk b/zishrink.awk
index 4e187ac..1947c7c 100644
--- a/zishrink.awk
+++ b/zishrink.awk
@@ -166,9 +166,6 @@ function process_input_line(line, \
   ruleline = sub(/^Rule /, "R ", line)
   zoneline = sub(/^Zone /, "Z ", line)
 
-  # SystemV rules are not needed.
-  if (line ~ /^R SystemV /) return
-
   # Replace FooAsia rules with the same rules without "Asia", as they
   # are duplicates.
   if (match(line, /[^ ]Asia /)) {
@@ -275,7 +272,6 @@ BEGIN {
   default_dep["factory"] = 1
   default_dep["northamerica"] = 1
   default_dep["southamerica"] = 1
-  default_dep["systemv"] = 1
   default_dep["ziguard.awk"] = 1
   default_dep["zishrink.awk"] = 1
 
-- 
1.8.3.1

